src/runtime/pkg/containerd-shim-v2/shim_management.go

```diff
@@ -241,6 +243,82 @@ func (s *service) genericIPTablesHandler(w http.ResponseWriter, r *http.Request,
 	}
 }
 
+type DeviceRequest struct {
+	DevicePath string
+}
+
+func (s *service) deviceHandler(w http.ResponseWriter, r *http.Request) {
+	logger := shimMgtLog.WithFields(logrus.Fields{"handler": "device"})
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		logger.WithError(err).Error("failed to read request body")
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(err.Error()))
+		return
+	}
+
+	switch r.Method {
+	case http.MethodGet:
```

Comment:
Not on this PR particularly, but we really need more abstraction in the shim management code, and a place to clearly define/document the APIs for each of these methods.

Reply:
I agree, IMO we should maybe hide the usage of http `PUT/POST/DELETE/GET` below some other functions like `Set/Update/Remove/Query` to clarify the meaning and easier to develop?